### ISSUE TITLE
[Feature] Add support for Tombs of Amascut

### ISSUE THREAD

**Issue by daltonpearson** [MosheBenZacharia/GP-Per-Hour#38]
Love the plugin! One of the most polished plugins out there. Also as a developer I'm happy to do the leg work on this issue and submit a PR I just want to get on the same page about your preferred implementation.

In ToA trips could behave slightly differently, the biggest pain-points are:
- using the deposit pot/friendly spirit(not sure which, I can do some testing) will reset the trip, this causes 2-3 trips to be created each raid and I think it would be more intuitive/useful if they were all treated as 1 trip.
- when using the send to bank option on the loot chest it doesn't count towards gains. I can see the argument for this being intended behavior but I don't see why anyone wouldn't want this included in their trip gains.
- support for splits would be nice but admittedly that may be better scoped to a raids specific loot plugin, unfortunately the current ones don't support ToA and I personally prefer to use your plugins UI.
- maybe add support to count death fees, could be a toggle
- would be nice to also include pre-pot in raid trip but this may be tricky

TLDR; Each raid should be considered one trip instead of the current 3.

Let me know what you think, if you think point 1 is out of scope I'll just roll it all into a new plugin. If you're open to these improvements we can strategize about implementation and I'll work on a PR if you like.

![image](https://github.com/MosheBenZacharia/GP-Per-Hour/assets/32880838/e0446d15-f0f6-436f-8050-fd65d559ecdc)

**Comment by MosheBenZacharia** [comment]
Hey, thanks for the detailed comment!

The workaround I added for the TOA pot was to get around the main issue with the implementation of this plugin which is that your profit/loss is calculated by comparing your inventory at the start of a trip to your inventory at the current point in time. To make this work every time the user encounters a 'bank' we need to start a trip and reset the 'start inventory' otherwise the calculations would be way off.

Ideally the plugin just calculates delta/differences each tick instead of keeping a record of each item in all your containers, which would simplify the code a lot. The reason it doesn't is because I built this on top of a different plugin which already had this system implemented and by the time I realized that it was going to cause major issues I was already about to release it. 

Anyways, the point is that it's a lot of work and I'm not actively developing this plugin anymore (besides for some small bug fixes here and there).

**If you're not terrified by the above and still want to go ahead and do this refactor I'd be happy to have you work on it, just let me know!**

There's a laundry list of high impact features/bug fixes that would only be possible to fix post refactor, I'll list them here in case you're curious:

        - [ ] [PostRe] Support payment for unlocking items from gravestone
        - [ ] [PostRe] Add support for loot keys
        - [ ] [PostRe] Option to not automatically start new trips when banking
        - [ ] [PostRe] Ability to right-click remove items from tracker (good for when you trade something)
        - [ ] [PostRe] Gauntlet support (all items count as a loss then as a gain again)
        - [ ] [PostRe] Giants foundry support (not tracking coins)
        - [ ] [PostRe] Support logging out and back in (refactor to be delta based)
        - [ ] [PostRe] Save entire active session instead of just active trip
        - [ ] [PostRe] Gold drop when switching from inventory total mode to anything else (need to keep separate totals)
        - [ ] [PostRe] Checking charges adds profit the first time

This refactor would also allow for solving #2, #4, and #5 from your post.

**Comment by MosheBenZacharia** [comment]
should be fixed when this PR merges: https://github.com/runelite/plugin-hub/pull/6153

**Comment by MosheBenZacharia** [comment]
should work now in version 1.5!
